### PR TITLE
Feature/skale 3756 Restructure mainnet contracts

### PR DESCRIPTION
**Pull request by DimaStebaev**

### CODE REVIEW

File: proxy/contracts/mainnet/Linker.sol [hunk 1]
```diff
@@ -21,92 +21,76 @@
...
-            }
-        }
-        return false;
+        return _depositBoxes.contains(depositBoxAddress);
     }
 
     function hasSchain(string calldata schainName) external view returns (bool connected) {
-        uint length = _depositBoxes.length;
+        uint length = _depositBoxes.length();
         connected = true;
```
Comment (payvint) [inline proxy/contracts/mainnet/Linker.sol]: Looks like we could simplify it: 
`connected = messageProxy.isConnectedChain(schainName)`
and then remove 
`connected = connected && messageProxy.isConnectedChain(schainName);`

File: proxy/contracts/mainnet/DepositBox.sol [hunk 2]
```diff
@@ -21,40 +22,49 @@
...
         require(msg.sender == address(messageProxy), "Sender is not a MessageProxy");
         _;
     }
 
     /**
-     * @dev initialize - sets current address of ContractManager
-     * @param newIMALinkerAddress - current address of ContractManager
+     * @dev Checks whether sender is owner of SKALE chain
      */
+    function isSchainOwner(address sender, bytes32 schainId) public view returns (bool) {
```
Comment (payvint) [inline proxy/contracts/mainnet/DepositBox.sol]: Maybe leave it in "SkaleManagerClient"?

**Review by payvint — CHANGES_REQUESTED** [review]
Good

**Review by payvint — APPROVED** [review]
Good!

**Review by sergiy-skalelabs — APPROVED** [review]

**Pull request merged**